R/utils_gcp.R

```diff
@@ -0,0 +1,102 @@
+# Semi-automated tests of Google Cloud Storage integration live in tests/gcp/. # nolint
+# These tests should not be fully automated because they
+# automatically create S3 buckets and upload data,
+# which could put an unexpected and unfair burden on
+# external contributors from the open source community.
+# nocov start
+gcp_gcs_exists <- function(
+  key, 
+  bucket = googleCloudStorageR::gcs_get_global_bucket(), 
+  version = NULL
+) {
+  tryCatch(
+    gcp_gcs_head_true(
+      key = key,
+      bucket = bucket,
+      version = version
+    ),
+    error = function(condition) {
+      FALSE
+    }
+  )
+}
+
+gcp_gcs_head <- function(
+  key, 
+  bucket = googleCloudStorageR::gcs_get_global_bucket(), 
+  version = NULL
+) {
+  suppressMessages(
+    googleCloudStorageR::gcs_get_object(
+      key,
+      bucket = bucket,
+      meta = TRUE,
+      generation = version
+    )
+  )
+
+}
+
+gcp_gcs_head_true <- function(
+  key, 
+  bucket = googleCloudStorageR::gcs_get_global_bucket(), 
+  version = NULL
+) {
+  gcp_gcs_head(
+    key = key,
+    bucket = bucket,
+    version = version
+  )
+  TRUE
+}
+
+gcp_gcs_download <- function(
+  file,
+  key,
+  bucket = googleCloudStorageR::gcs_get_global_bucket(),
+  version = NULL
+) {
+  
+  googleCloudStorageR::gcs_get_object(
+    key,  
+    bucket = bucket,
+    saveToDisk = file,
+    overwrite = TRUE,
+    generation = version
+  )
+
+}
+
+
+gcp_gcs_upload <- function(
+  file,
+  key,
+  bucket = googleCloudStorageR::gcs_get_global_bucket(),
+  metadata = list(),
+  predefinedAcl = c("private", "bucketLevel", "authenticatedRead",
```

Comment:
A couple minor style requests:

1. `targets` tries to use snake case, so would you be willing to call this argument `predefined_acl`? 
2. Would you be willing to start the elements of default argument vector on a new line indented by 2 spaces? Picturing something like this:


```r
predefined_acl = c(
  "private", "bucketLevel", "authenticatedRead",
  "projectPrivate", "publicRead", "default"
)
```

I try to follow https://style.tidyverse.org/syntax.html#long-lines.

Reply:
Sure, the camelCase is to keep in line with the API field name but I usually default to snake_case when I can.